--- src/server/kiwix-serve.cpp	2017-01-25 04:01:05.000000000 +0000
+++ src/server/kiwix-serve.cpp	2017-01-25 05:17:28.165863903 +0000
@@ -241,7 +241,7 @@
 {
   zim::Blob* blob = static_cast<zim::Blob*>(cls);
   pthread_mutex_lock(&readerLock);
-  size_t max_size_to_set = min(max, blob->size()-pos);
+  size_t max_size_to_set = min<size_t>(max, blob->size()-pos);
 
   if (max_size_to_set <= 0)
   {
